arcaflow_plugin_stressng/stressng_schema.py

```diff
@@ -25,176 +197,376 @@ class CommonStressorParams:
-                "reading and removing temporary files"
-            ),
-        }
-    )
-
-    hdd_bytes: str = field(
-        metadata={
-            "name": "Bytes per worker",
-            "description": (
-                "write  N  bytes for each hdd process, the default is 1 GB. "
-                "One can specify the size in units of Bytes, KBytes, "
-                "MBytes and GBytes using the suffix b, k, m or g."
-            ),
-        }
-    )
-
-    hdd_write_size: str = field(
-        metadata={
-            "name": "Write Size",
-            "description": (
-                "specify size of each write "
-                "in bytes. Size can be from 1 byte to 4MB"
-                "One can specify the size in units of Bytes, KBytes, "
-                "MBytes using the suffix b, k, m"
-            ),
-        }
-    )
+    hdd_bytes: typing.Annotated[
+        typing.Optional[str],
+        schema.id("hdd-bytes"),
+        schema.name("Bytes Per Worker"),
+        schema.description("Write N bytes for each hdd process; the default is 1 GB"),
+    ] = None
+
+    hdd_opts: typing.Annotated[
+        typing.Optional[typing.List[HddOpts]],
+        schema.id("hdd-opts"),
+        schema.name("HDD Options"),
+        schema.description("Various stress test options as a list"),
+    ] = None
+
+    hdd_ops: typing.Annotated[
+        typing.Optional[int],
+        schema.id("hdd-ops"),
+        schema.name("HDD Operations"),
+        schema.description("Stop hdd stress workers after N bogo operations"),
+    ] = None
+
+    hdd_write_size: typing.Annotated[
+        typing.Optional[str],
+        schema.id("hdd-write-size"),
+        schema.name("HDD Write Size"),
+        schema.description("Size of each write in bytes"),
+    ] = None
 
     def to_jobfile(self) -> str:
-        hdd = "hdd {}\n".format(self.hdd)
-        hdd_bytes = "hdd-bytes {}\n".format(self.hdd_bytes)
-        hdd_write_size = "hdd-write-size {}\n".format(self.hdd_write_size)
-        result = hdd + hdd_bytes + hdd_write_size
+        params = schema.build_object_schema(HDDStressorParams).serialize(
+            HDDStressorParams(
+                Stressors(self.stressor),
+                self.workers,
+                self.hdd_bytes,
+                self.hdd_opts,
+                self.hdd_ops,
+                self.hdd_write_size,
+            )
+        )
+        result = f"hdd {self.workers}\n"
+        result += params_to_jobfile(params)
         return result
 
 
 @dataclass
 class StressNGParams:
-    """
-    The parameters in this schema will be passed through to the stressng
-    command unchanged
-    """
-
-    timeout: str = field(
-        metadata={
-            "name": "Runtime",
-            "description": "Time to run the benchmark test",
-        }
-    )
+    timeout: typing.Annotated[
+        int,
+        schema.name("Timeout"),
+        schema.description("Stop stress test after T seconds"),
```

Comment:
Again, not crazy about the "T" seconds.

But, also, it's interesting that you've "simplified" by changing the timeout value to `int` instead of allowing the full expressiveness of adding units which was implied by the previous type, e.g., `10m` vs `600`. Is this somehow necessary? Or was the old code just wrong?

Reply:
Dave, there's another possibility:  `10m` could actually _be_ an `int`!  I mean, if `1K` means `1000`, then, when the parser is done with it, `10m` could mean `600`...it all depends on where in the parsing the schema is applied....  🙃

Reply:
I think you're suggesting that ARCA is coercing a string value of "10m" into an integer value of 600 because the schema identifies `timeout` as an `int`?? I'm pretty sure you'd be disappointed...

Reply:
True enough...Arca doesn't know about time values...yet....  😈

Reply:
The underlying stress-ng tool allows either integers or strings with unit notations. The stress-ng docs, however, explicitly describe this as seconds, so I felt it was more protective of the plugin user to set our own schema to only accept integer seconds. Accepting unit notations while also validating input at the plugin (so we don't just wait for the underlying stress-ng to choke on it) means writing string validation, and to me it just makes more sense to leave this as an integer.